### PR TITLE
Bug/remove pupil census user defaults

### PR DESCRIPTION
**Pull request by jon-shipley**
Remove default on the census upload user password

Adds:
* new schema for census upload to use for staging tables
* deny perms for the census upload user to see the mtc_admin schema tables except for 'pupil' which  has limited permissions (can't select)

### CODE REVIEW

File: admin/data/sql/migrations/20190320145618.do.grant-census-import-user-permissions.js [hunk 1]
```diff
@@ -3,12 +3,13 @@
 const config = require('../../../config')
 
 module.exports.generateSql = function () {
-  if (config.Sql.Azure.Scale) {
-    return `
+  return `
+    CREATE SCHEMA [mtc_census_import];
+    GO    
+    GRANT CONTROL ON schema::[mtc_census_import] TO [${config.Sql.PupilCensus.Username}];
```
Comment (GeorgeChatzigiannis) [inline admin/data/sql/migrations/20190320145618.do.grant-census-import-user-permissions.js]: @jon-shipley should there be a revoke on that on the undo statement?

File: admin/data/sql/migrations/20190320145618.do.grant-census-import-user-permissions.js [hunk 2]
```diff
@@ -3,12 +3,13 @@
 const config = require('../../../config')
 
 module.exports.generateSql = function () {
-  if (config.Sql.Azure.Scale) {
-    return `
+  return `
+    CREATE SCHEMA [mtc_census_import];
```
Comment (GeorgeChatzigiannis) [inline admin/data/sql/migrations/20190320145618.do.grant-census-import-user-permissions.js]: @jon-shipley shouldn't you undo the schema creation on the undo statement?

Comment (jon-shipley) [reply admin/data/sql/migrations/20190320145618.do.grant-census-import-user-permissions.js]: You were too fast!  Now added.

File: admin/data/sql/migrations/20190320145000.do.create-census-import-user.js [hunk 3]
```diff
@@ -3,12 +3,19 @@
 const config = require('../../../config')
 
 const createAzureUser = `CREATE USER ${config.Sql.PupilCensus.Username} WITH PASSWORD ='${config.Sql.PupilCensus.Password}', DEFAULT_SCHEMA=[mtc_admin];`
+const createLocalUser = `CREATE LOGIN ${config.Sql.PupilCensus.Username} WITH PASSWORD = '${config.Sql.PupilCensus.Password}'; USE ${config.Sql.Database}; CREATE USER ${config.Sql.PupilCensus.Username} FOR LOGIN ${config.Sql.PupilCensus.Username} WITH DEFAULT_SCHEMA = [mtc_admin];`
 
 // TODO test on sql azure
 module.exports.generateSql = function () {
+  if (!config.Sql.PupilCensus.Username) {
+    throw new Error(`Missing 'config.Sql.PupilCensus.Username'`)
```
Comment (GuyHarwood) [inline admin/data/sql/migrations/20190320145000.do.create-census-import-user.js]: 💯

**Comment by GuyHarwood** [comment]
@jon-shipley [PR-20190322.7 build](https://agilefactory.visualstudio.com/MTC/_build/results?buildId=33953) passed

**Comment by GuyHarwood** [comment]
@jon-shipley travis.yml needs the env var for census password to be added for it to pass

**Pull request merged**